robot/Cumulus/resources/RecurringDonationsPageObject.py

```diff
@@ -87,7 +87,18 @@ def click_actions_button(self, button_name):
             self.selenium.click_link(button_name)
         else:
             self.selenium.click_button(button_name)
+            
+            
+    def go_to_recurring_donation_related_opportunities_page(self,rd_id):
 
+        """ Navigates to the related opportunities page of the given recurring donation """
+        objectname = "npe03__Donations__r"
+        values =  self.npsp.get_url_formatted_object_name(objectname)
+        url = "{}/lightning/r/{}/related/{}/view".format(values['baseurl'],rd_id,objectname)
+        self.selenium.go_to(url)
+        self.salesforce.wait_until_loading_is_complete()
+        self.selenium.wait_until_location_contains("/view",timeout=60, message="Page not loaded")
```

Comment:
Just curious to know if this always lets us know that the opportunity record has loaded successfully? Can we use any other data from the oppty itself to validate that the record is available to test?

Reply:
yup agree with rakesh, I recommend waiting for something to be visible on the page to confirm page has loaded and ready to test